### PR TITLE
Add organisation field

### LINKED ISSUE #59: Add organisation field

### PR DESCRIPTION
**Pull request by IanCheah**
Fixes #59

### CODE REVIEW

File: src/test/java/seedu/address/model/person/OrganisationTest.java
Comment (nigellqx) [inline src/test/java/seedu/address/model/person/OrganisationTest.java:1]: LGTM. All test cases are covered.

File: src/test/java/seedu/address/model/person/OrganisationTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    @Test
+    public void constructor_invalidOrganisation_throwsIllegalArgumentException() {
+        String invalidOrganisation = "";
+        assertThrows(IllegalArgumentException.class, () -> new Organisation(invalidOrganisation));
+    }
+
+    @Test
+    public void isValidOrganisation() {
+        // null address
```
Comment (nigellqx) [inline src/test/java/seedu/address/model/person/OrganisationTest.java:24]: I think you mean "null organisation".

File: src/test/java/seedu/address/model/person/OrganisationTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        assertThrows(NullPointerException.class, () -> Organisation.isValidOrganisation(null));
+
+        // invalid addresses
```
Comment (nigellqx) [inline src/test/java/seedu/address/model/person/OrganisationTest.java:27]: I think you mean "invalid organisation".

File: src/test/java/seedu/address/model/person/OrganisationTest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        assertFalse(Organisation.isValidOrganisation("")); // empty string
+        assertFalse(Organisation.isValidOrganisation(" ")); // spaces only
+
+        // valid addresses
```
Comment (nigellqx) [inline src/test/java/seedu/address/model/person/OrganisationTest.java:31]: I think you mean "valid organisation".

File: src/test/java/seedu/address/testutil/PersonBuilder.java [hunk 4]
```diff
@@ -88,9 +93,16 @@ public PersonBuilder withEmail(String email) {
         this.email = new Email(email);
         return this;
     }
+    /**
+     * Sets the {@code Email} of the {@code Person} that we are building.
```
Comment (nigellqx) [inline src/test/java/seedu/address/testutil/PersonBuilder.java:97]: I think your java docs meant organization here.

File: src/test/java/seedu/address/model/person/OrganisationTest.java (hunk 1 above)
Comment (IanCheah) [reply src/test/java/seedu/address/model/person/OrganisationTest.java]: yes, I meant "null organisation". Good find.

File: src/test/java/seedu/address/model/person/OrganisationTest.java (hunk 2 above)
Comment (IanCheah) [reply src/test/java/seedu/address/model/person/OrganisationTest.java]: yes, I meant "invalid organisation". Good find.

File: src/test/java/seedu/address/model/person/OrganisationTest.java (hunk 3 above)
Comment (IanCheah) [reply src/test/java/seedu/address/model/person/OrganisationTest.java]: yes, I meant "valid organisation". Good find.

File: src/test/java/seedu/address/testutil/PersonBuilder.java (hunk 4 above)
Comment (IanCheah) [reply src/test/java/seedu/address/testutil/PersonBuilder.java]: yes, I meant organization. Good find.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 5]
```diff
@@ -95,6 +96,21 @@ public static Email parseEmail(String email) throws ParseException {
...
+    /**
+     * Parses a {@code String organisation} into an {@code Organisation}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code organisation} is invalid.
+     */
+    public static Organisation parseOrganisation(String organisation) throws ParseException {
+        requireNonNull(organisation);
+        String trimmedOrganisation = organisation.trim();
+        if (!Address.isValidAddress(trimmedOrganisation)) {
```
Comment (YINZE66) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:108]: Find this bug when deleting the address field. Should be isValidOrganisation here.

**Pull request merged**